Core/Core/Transports/ITransport.cs

```diff
@@ -66,16 +66,6 @@ public interface ITransport
   /// <exception cref="OperationCanceledException"><see cref="CancellationToken"/> requested cancel</exception>
   public void SaveObject(string id, string serializedObject);
 
-  /// <summary>
-  /// <inheritdoc cref="SaveObject(string, string)"/>
-  /// Retrieving its serialised version from the provided transport.
-  /// </summary>
-  /// <param name="id"><inheritdoc cref="SaveObject(string, string)"/></param>
-  /// <param name="sourceTransport">The transport from where to retrieve it.</param>
-  /// <exception cref="TransportException">Failed to save object</exception>
-  /// <exception cref="OperationCanceledException"><see cref="CancellationToken"/> requested cancel</exception>
-  public void SaveObject(string id, ITransport sourceTransport);
```

Comment:
This is the one that was moved to a shared implementation right?

Not sure how many people would have their own transport but technically this is a breaking change if your transport already has `SaveObject` implemented right? But also we're saying a transport "doesn't save objects" when that's it's whole point of existing.

Since the extension lives in the same namespace as the interface, it must cause a conflict. I'm not sure we should remove this from here, though the shared implementation should stay.

It would be a bit more verbose to just keep this and call the shared implementation on every transport class.

Reply:
There shouldn't be any conflicts. If you call `SaveObject` on an `ITransport`, it'll use the extension method. If you call `SaveObject` on your concrete type, then it will use the member function.
Case in point, I didn't touch `DiskTransport`, and the tests still pass! (side note: should I update `DiskTransport` too? )

The only potential issues would be if somehow you're relying on some custom behaviour in `SaveObject(string, ITransport)`

If you think that's possible, then I can revert this change.

> But also we're saying a transport "doesn't save objects" when that's it's whole point of existing.

Not sure I quite follow here... maybe I just revert and we chat in the new year?

Reply:
I meant that the interface is in essence our contract to other developers detailing what a class "should do", even if all our Transport classes do the same thing, I don't think it's safe to say that "every transport" will do the same?

We have no knowledge of other transports existing but that doesn't mean they don't 😅

I would revert the deletion in the interface and just do

`public void SaveObject(...) => TransportExtensions.SaveObject(...)`

Obviously `TransortExtensions` naming would no longer make sense and may need revisiting... maybe `TransportUtilities` or `TransportBase`? 🤔 Naming is hard...

This way `TransportExtensions.SaveObject` doesn't need to be an extension method either, and we don't break the existing contract (even though I'm convinced after your comment it is not really breaking)

Reply:
I've reverted the `ITransport` change, and will include it in a separate PR later. We can discuss that more when you're back!